### PR TITLE
Create k8s service and expose container ports for consensus in allreduce training

### LINKED ISSUE #1710: Create k8s service that exposes FTLib's consensus TCP connection

### PR DESCRIPTION
**Pull request by terrytangyuan**
Fixes https://github.com/sql-machine-learning/elasticdl/issues/1710.

Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

### CODE REVIEW

File: elasticdl/python/common/k8s_client.py [hunk 1]
```diff
@@ -264,6 +266,14 @@ def _create_pod(self, **kargs):
             args=kargs["container_args"],
             image_pull_policy=kargs["image_pull_policy"],
             env=kargs["env"],
+            ports=[
```
Comment (skydoorkai) [inline elasticdl/python/common/k8s_client.py]: These two ports will be added for all pods, including PS which does not use them.

Comment (terrytangyuan) [reply elasticdl/python/common/k8s_client.py]: Good catch. I've updated the PR to address this. Please take another look.

File: elasticdl/python/common/k8s_client.py [hunk 2]
```diff
@@ -367,19 +385,23 @@ def _create_ps_worker_pod(self, pod_name, type_key, index_key, **kargs):
     def create_worker(self, **kargs):
         pod_name = self.get_worker_pod_name(kargs["worker_id"])
         return self._create_ps_worker_pod(
-            pod_name, "worker", kargs["worker_id"], **kargs
+            pod_name, "worker", kargs["worker_id"], expose_ports=True, **kargs
```
Comment (skydoorkai) [inline elasticdl/python/common/k8s_client.py]: For worker pods, `expose_ports=True` only in DistributionStrategy.ALLREDUCE, not in DistributionStrategy.PARAMETER_SERVER.

Comment (terrytangyuan) [reply elasticdl/python/common/k8s_client.py]: Thanks. Addressed.

**Review by skydoorkai — APPROVED** [review]

**Pull request merged**